mlir/lib/Conversion/MPIToLLVM/MPIToLLVM.cpp

```diff
@@ -0,0 +1,501 @@
+  static constexpr int MPI_INT32_T = 0x4c000439;
+  static constexpr int MPI_INT64_T = 0x4c00083a;
+  static constexpr int MPI_UINT8_T = 0x4c00013b;
+  static constexpr int MPI_UINT16_T = 0x4c00023c;
+  static constexpr int MPI_UINT32_T = 0x4c00043d;
+  static constexpr int MPI_UINT64_T = 0x4c00083e;
+
+public:
+  using MPIImplTraits::MPIImplTraits;
+
+  Value getCommWorld(const Location loc,
+                     ConversionPatternRewriter &rewriter) override {
+    static const int MPI_COMM_WORLD = 0x44000000;
+    return rewriter.create<LLVM::ConstantOp>(loc, rewriter.getI32Type(),
+                                             MPI_COMM_WORLD);
+  }
+
+  intptr_t getStatusIgnore() override { return 1; }
+
+  Value getDataType(const Location loc, ConversionPatternRewriter &rewriter,
+                    Type type) override {
+    int32_t mtype = 0;
+    if (type.isF32())
+      mtype = MPI_FLOAT;
+    else if (type.isF64())
+      mtype = MPI_DOUBLE;
+    else if (type.isInteger(64) && !type.isUnsignedInteger())
+      mtype = MPI_INT64_T;
+    else if (type.isInteger(64))
+      mtype = MPI_UINT64_T;
+    else if (type.isInteger(32) && !type.isUnsignedInteger())
+      mtype = MPI_INT32_T;
+    else if (type.isInteger(32))
+      mtype = MPI_UINT32_T;
+    else if (type.isInteger(16) && !type.isUnsignedInteger())
+      mtype = MPI_INT16_T;
+    else if (type.isInteger(16))
+      mtype = MPI_UINT16_T;
+    else if (type.isInteger(8) && !type.isUnsignedInteger())
+      mtype = MPI_INT8_T;
+    else if (type.isInteger(8))
+      mtype = MPI_UINT8_T;
+    else
+      assert(false && "unsupported type");
+    return rewriter.create<LLVM::ConstantOp>(loc, rewriter.getI32Type(), mtype);
+  }
+};
+
+//===----------------------------------------------------------------------===//
+// Implementation details for OpenMPI
+//===----------------------------------------------------------------------===//
+class OMPIImplTraits : public MPIImplTraits {
+  LLVM::GlobalOp getOrDefineExternalStruct(const Location loc,
+                                           ConversionPatternRewriter &rewriter,
+                                           StringRef name,
+                                           LLVM::LLVMStructType type) {
+
+    return getOrDefineGlobal<LLVM::GlobalOp>(
+        getModuleOp(), loc, rewriter, name, type, /*isConstant=*/false,
+        LLVM::Linkage::External, name,
+        /*value=*/Attribute(), /*alignment=*/0, 0);
+  }
+
+public:
+  using MPIImplTraits::MPIImplTraits;
+
+  Value getCommWorld(const Location loc,
+                     ConversionPatternRewriter &rewriter) override {
+    auto context = rewriter.getContext();
+    // get external opaque struct pointer type
+    auto commStructT =
+        LLVM::LLVMStructType::getOpaque("ompi_communicator_t", context);
+    StringRef name = "ompi_mpi_comm_world";
+
+    // make sure global op definition exists
+    (void)getOrDefineExternalStruct(loc, rewriter, name, commStructT);
```

Comment:
```suggestion
    getOrDefineExternalStruct(loc, rewriter, name, commStructT);
```
Nit: Should not be necessary, as the function is not marked with `nodiscard`